plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt

```diff
@@ -394,17 +401,23 @@ class CodeModernizerSession(
             throw AlreadyDisposedException("Disposed when about to upload project artifact to s3")
         }
         val uploadStartTime = Instant.now()
-        try {
+        waitUntil (
+            exceptionsToIgnore = setOf(
+                UnknownHostException::class,
+                SocketTimeoutException::class,
+                HttpRequests.HttpStatusException::class,
+                ConnectException::class
```

Comment:
`UnknownHostException` and `SocketTimeoutException` can be reproduced by turning off your WiFi, `HttpStatusException` could be a `500` or something, and I've occasionally seen `ConnectException` too, so I think all of these are worth retrying on.